Tests/TestUtilities/AssertException.cs

```diff
@@ -51,6 +49,21 @@ public static class AssertException
             return (TException)caught;
         }
 
-        #endregion Public methods
+        public static void ExpectsNullArgumentException(Action op, string parameterName)
```

Comment:
I don't think we need another way of testing for argument null expections: we already have to two patterns in common use:
* _AssertException.Expects<ArgumentNullException>(() => testSubject.CreateSonarQubeServer(null, logger));_
* the FluentAssertions approach you use in the _ProjectInfoExtensionsTests_ below

If you think it's worth adding a specific helper method, it might be better to make it a thin wrapper around the FluentAssertions pattern.